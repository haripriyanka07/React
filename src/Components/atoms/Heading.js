export default function Heading(props) {
    return (
        <h1 style={props.style}>{props.value}</h1>
    )
}
export function Hseading(props) {
    return (
        <h2>{props.value}</h2>
    )
}
