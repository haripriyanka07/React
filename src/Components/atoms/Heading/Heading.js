export default function Heading(props) {
    return (
        <h1 data-testid="h1tag" style={props.style}>{props.value}</h1>
    )
}

