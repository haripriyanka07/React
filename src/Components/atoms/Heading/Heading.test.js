import React from 'react';
import Heading from './Heading';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";
afterEach(cleanup);

// describe('test the Heading', () => {
    // it('should return the value passed as header', async() => {
    //     const { getByText } = render(<Heading value="Heading" />);
    //     // console.log(screen)
    //     expect(getByText(/Heading/))
    //     var component = TestUtils.renderIntoDocument(
    //         <Heading />
    //      );
    //     var h1 = TestUtils.findRenderedDOMComponentWithTag(
    //         component, 'h1'
    //      );
    //      console.log(h1)
    // })
    // it("renders an h1", function () {
    //     var component = render(
    //         <Heading value="Hari"/>
    //     );
    //     console.log(component)
    //     console.log(component.getByLabelText())
    
    //     var h1 = TestUtils.findRenderedDOMComponentWithTag(
    //        component, 'h1'
    //     );
    
    //     expect(h1.getDOMNode().textContent)
    //         .toEqual("Hari");
    // });

    

// })

// test("basic snapshot", () => {
    // const { getByTestId } = render(<Heading value="Contact" />);
    // expect(getByTestId("h1tag")).toHaveTextContent("Contact");
//   });

it("render", () => {
    const { asFragment } = render(<Heading value="cruze" />);
    expect(asFragment()).toMatchSnapshot();
})
it('should return the value passed as header', async() => {
    const { getByTestId } = render(<Heading value="Contact" />);
    expect(getByTestId("h1tag")).toHaveTextContent("Contact");
})