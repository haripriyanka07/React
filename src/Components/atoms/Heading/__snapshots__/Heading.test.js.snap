// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<DocumentFragment>
  <h1
    data-testid="h1tag"
  >
    cruze
  </h1>
</DocumentFragment>
`;
